integration_tests/test_store_sign_build.py

```diff
@@ -51,17 +53,27 @@ def test_unsuccessful_sign_build_no_login(self):
         self.assertThat(self.snap_build_path, Not(FileExists()))
 
     def test_successful_sign_build_local(self):
-        self.run_snapcraft('snap', self.project_dir)
-        os.chdir(self.project_dir)
-        self.assertThat(self.snap_path, FileExists())
-
         self.addCleanup(self.logout)
         self.login()
 
-        status = self.sign_build(self.snap_path, local=True)
+        # Build a random snap and register it.
+        unique_id = uuid.uuid4().int
+        name = 'u1test-{}'.format(unique_id)
+        version = str(unique_id)[:32]
+        project_dir = self.update_name_and_version(
+            self.project_dir, name, version)
+        os.chdir(project_dir)
+        self.run_snapcraft('snap', project_dir)
+        snap_path = '{}_{}_{}.snap'.format(name, version, 'all')
+        self.assertThat(snap_path, FileExists())
+        self.register(name)
+
+        status = self.sign_build(snap_path, local=True)
         self.assertEqual(0, status)
-        self.assertThat(self.snap_build_path, FileExists())
-        self.assertThat(self.snap_build_path,
+
+        snap_build_path = '{}-build'.format(snap_path)
```

Comment:
It was, the test works if the account logged in was the publisher of 'basic' name, but this is impossible in staging SCA. See https://travis-ci.org/snapcore/snapcraft/builds/163606844